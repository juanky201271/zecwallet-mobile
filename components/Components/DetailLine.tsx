/* eslint-disable react-native/no-inline-styles */
import React, { ReactNode, useContext } from 'react';
import { TouchableOpacity, View } from 'react-native';
import Clipboard from '@react-native-community/clipboard';
import { useTheme } from '@react-navigation/native';

import FadeText from './FadeText';
import RegText from './RegText';
import { ThemeType } from '../../app/types';
import { ContextAppLoaded } from '../../app/context';
import moment from 'moment';
import 'moment/locale/es';
import 'moment/locale/pt';
import 'moment/locale/ru';
import { SnackbarDurationEnum } from '../../app/AppState';

type DetailLineProps = {
  label: string;
  value?: string;
  children?: ReactNode;
  testID?: string;
};

const DetailLine: React.FunctionComponent<DetailLineProps> = ({ label, value, children, testID }) => {
  const { colors } = useTheme() as unknown as ThemeType;
  const context = useContext(ContextAppLoaded);
  const { addLastSnackbar, translate, language } = context;
  moment.locale(language);

  return (
    <View style={{ display: 'flex', marginTop: 20 }}>
      <FadeText>{label}</FadeText>
      {!!value && (
        <TouchableOpacity
          onPress={() => {
            Clipboard.setString(value);
            addLastSnackbar({
              message: translate('txtcopied') as string,
              duration: SnackbarDurationEnum.short,
            });
          }}>
          <RegText testID={testID} color={colors.text}>
            {value}
          </RegText>
        </TouchableOpacity>
      )}
      {children}
    </View>
  );
};

export default DetailLine;
