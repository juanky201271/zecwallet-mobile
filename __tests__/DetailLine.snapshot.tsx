/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import DetailLine from '../components/Components/DetailLine';
import { View } from 'react-native';

// test suite
describe('Component DetailLine - test', () => {
  //snapshot test
  test('DetailLine value - snapshot', () => {
    const detail = render(<DetailLine label="label" value="value" />);
    expect(detail.toJSON()).toMatchSnapshot();
  });
  test('DetailLine children - snapshot', () => {
    const children = <View />;
    const detail = render(<DetailLine label="label" children={children} />);
    expect(detail.toJSON()).toMatchSnapshot();
  });
});
