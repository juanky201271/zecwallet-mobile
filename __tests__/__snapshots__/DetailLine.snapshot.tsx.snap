// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component DetailLine - test DetailLine children - snapshot 1`] = `
<View
  style={
    {
      "display": "flex",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "rgb(28, 28, 30)",
        "opacity": 0.65,
      }
    }
  >
    label
  </Text>
  <View />
</View>
`;

exports[`Component DetailLine - test DetailLine value - snapshot 1`] = `
<View
  style={
    {
      "display": "flex",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "rgb(28, 28, 30)",
        "opacity": 0.65,
      }
    }
  >
    label
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(28, 28, 30)",
          "fontSize": 18,
          "fontWeight": "600",
          "opacity": 1,
        }
      }
    >
      value
    </Text>
  </View>
</View>
`;
